File: backend/src/main/java/com/cms/module/site/entity/Site.java
package com.cms.module.site.entity;

import com.cms.common.base.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.persistence.*;

/**
 * 站点实体
 *
 * @author CMS Team
 * @since 1.0.0
 */
@Data
@EqualsAndHashCode(callSuper = true)
@Entity
@Table(name = "sites")
public class Site extends BaseEntity {

    /**
     * 站点名称
     */
    @Column(nullable = false, length = 100)
    private String name;

    /**
     * 站点代码
     */
    @Column(nullable = false, unique = true, length = 50)
    private String code;

    /**
     * 站点域名
     */
    @Column(nullable = false, length = 255)
    private String domain;

    /**
     * 站点描述
     */
    @Column(columnDefinition = "TEXT")
    private String description;

    /**
     * Logo URL
     */
    @Column(name = "logo_url", length = 500)
    private String logoUrl;

    /**
     * Favicon URL
     */
    @Column(name = "favicon_url", length = 500)
    private String faviconUrl;

    /**
     * 默认语言
     */
    @Column(length = 10)
    private String language = "zh_CN";

    /**
     * 时区
     */
    @Column(length = 50)
    private String timezone = "Asia/Shanghai";

    /**
     * 站点状态
     */
    @Column(nullable = false, columnDefinition = "ENUM('ACTIVE', 'INACTIVE', 'MAINTENANCE') DEFAULT 'ACTIVE'")
    private String status = "ACTIVE";

    /**
     * 是否默认站点
     */
    @Column(name = "is_default", nullable = false)
    private Boolean isDefault = false;

    /**
     * SEO标题
     */
    @Column(name = "seo_title", length = 200)
    private String seoTitle;

    /**
     * SEO关键词
     */
    @Column(name = "seo_keywords", length = 500)
    private String seoKeywords;

    /**
     * SEO描述
     */
    @Column(name = "seo_description", columnDefinition = "TEXT")
    private String seoDescription;

    /**
     * 联系邮箱
     */
    @Column(name = "contact_email", length = 100)
    private String contactEmail;

    /**
     * 联系电话
     */
    @Column(name = "contact_phone", length = 20)
    private String contactPhone;

    /**
     * 联系地址
     */
    @Column(name = "contact_address", length = 500)
    private String contactAddress;
}

File: frontend/src/locales/en-US/site.ts
/**
 * English Language Pack - Site Module
 */
export default {
  title: 'Site Management',
  list: 'Site List',
  add: 'Create Site',
  edit: 'Edit Site',
  delete: 'Delete Site',
  detail: 'Site Details',
  
  fields: {
    id: 'ID',
    name: 'Site Name',
    code: 'Site Code',
    domain: 'Domain',
    description: 'Description',
    status: 'Status',
    language: 'Language',
    timezone: 'Timezone',
    theme: 'Theme',
    logo: 'Logo',
    favicon: 'Favicon',
    seoTitle: 'SEO Title',
    seoKeywords: 'SEO Keywords',
    seoDescription: 'SEO Description',
    contactEmail: 'Contact Email',
    contactPhone: 'Contact Phone',
    contactAddress: 'Contact Address',
    isDefault: 'Default Site',
    createdAt: 'Created At',
    updatedAt: 'Updated At'
  },
  
  status: {
    active: 'Active',
    inactive: 'Inactive',
    maintenance: 'Maintenance',
    ACTIVE: 'Running',
    INACTIVE: 'Disabled'
  },
  
  placeholder: {
    name: 'Enter site name',
    code: 'Enter site code (lowercase letters, numbers, underscores, hyphens)',
    domain: 'Enter site domain',
    description: 'Enter site description',
    searchKeyword: 'Search site name or domain',
    selectStatus: 'Select status',
    selectLanguage: 'Select default language',
    selectTimezone: 'Select timezone',
    selectSite: 'Select Site',
    seoTitle: 'Enter SEO title',
    seoKeywords: 'Enter SEO keywords, separated by commas',
    seoDescription: 'Enter SEO description',
    contactEmail: 'Enter contact email',
    contactPhone: 'Enter contact phone',
    contactAddress: 'Enter contact address',
    customCss: 'Enter custom CSS code',
    customJs: 'Enter custom JavaScript code'
  },
  
  validation: {
    nameRequired: 'Site name is required',
    nameLength: 'Site name must be 1-100 characters',
    nameMaxLength: 'Site name cannot exceed 100 characters',
    codeRequired: 'Site code is required',
    codeMaxLength: 'Site code cannot exceed 50 characters',
    codePattern: 'Site code can only contain lowercase letters, numbers, underscores and hyphens',
    domainRequired: 'Site domain is required',
    domainMaxLength: 'Site domain cannot exceed 255 characters',
    domainInvalid: 'Invalid domain format',
    siteRequired: 'Please select a site'
  },
  
  message: {
    addSuccess: 'Site created successfully',
    addFailed: 'Failed to create site',
    updateSuccess: 'Site updated successfully',
    updateFailed: 'Failed to update site',
    deleteSuccess: 'Site deleted successfully',
    deleteFailed: 'Failed to delete site',
    deleteConfirm: 'Are you sure you want to delete site "{name}"?',
    loadFailed: 'Failed to load site list',
    createSuccess: 'Created successfully',
    saveFailed: 'Failed to save site',
    configSaveSuccess: 'Configuration saved successfully',
    configSaveFailed: 'Failed to save configuration',
    logoUploadSuccess: 'Logo uploaded successfully',
    faviconUploadSuccess: 'Favicon uploaded successfully'
  },

  // Additional content
  addSite: 'Add Site',
  config: 'Config',
  empty: 'No site data',
  noDescription: 'No description',
  defaultSite: 'Default Site',

  // Configuration related
  siteConfig: 'Site Configuration',
  basicConfig: 'Basic Configuration',
  themeConfig: 'Theme Configuration',
  advancedConfig: 'Advanced Configuration',
  primaryColor: 'Primary Color',
  secondaryColor: 'Secondary Color',
  customCss: 'Custom CSS',
  customJs: 'Custom JS',
  saveConfig: 'Save Configuration',

  // SEO settings
  seoSettings: 'SEO Settings',
  contactInfo: 'Contact Information',

  // Language options
  languages: {
    zh_CN: 'Simplified Chinese',
    en_US: 'English',
    zh_TW: 'Traditional Chinese'
  }
}

